### PR TITLE
Adds a docker image for the domain decomp tool

### LINKED ISSUE #69: add docker image of domain decomp tool
It would be useful to have a dockerized version of `decomp` that could be run to generate the inputs without needing to install dependencies and build the source.

### PR DESCRIPTION
**Pull request by einola**
The point is that a user can use the tool to partition a given grid without going to the hassle of installing Zoltan and other prerequisites. It works well enough as is, even if the syntax is not the most obvious (but I added a section in the README.md file).

The only sore point is that I used ``COPY . /decomp`` in the Dockerfile, which triggers a rebuild of the tool every time the Docker image is built after modifying any file in the root directory. The alternatives are to select which files to copy or to pull everything from GitHub. Pulling from GitHub seems a bit much since the source is already there, and copying select files means updating the list if the project changes. So I went for what I considered the least bad solution.

Closes #69

### CODE REVIEW

**Comment by einola** [comment]
I needed this, so I did it :)

**Review by joewallwork — CHANGES_REQUESTED** [review]
Nice, thanks for this @einola! I like your use of an entrypoint script. I guess another natural (follow-up) extension to this PR would be to have the nextSIM-DG Docker images(s) pull the `decomp` installation from the one created here.

I hit a build error when I tried to run locally because it was picking up a different `decomp` binary in the `build` directory. `rm -rf build` did the trick and with that the Dockerised tool works nicely.

To avoid others hitting the same issue, perhaps we should put the Dockerfile in a subdirectory (e.g., `Dockerfiles/`), as we did in the nextSIM-DG? Alternatively, the setup could be left as-is, but the gotcha above would be a useful addition to the docs.

File: README.md
Comment (einola) [reply README.md:302]: Yes, we should build a hierarchy of docker images on nextsimhub. That should be a separate issue, though. Also, I'm using an M1 Mac, which means I can't use x64 docker images like the ones on nextsimhub.

Comment (joewallwork) [reply README.md:302]: Ah okay, I hadn't thought about that.

**Review by joewallwork — APPROVED** [review]
Thanks @einola, this should be good to go once you merge in `main` to pick up the Mac CI fix.

**Pull request merged**
